### PR TITLE
Dev/card detail

### PR DESCRIPTION
**Pull request by jdstmporter**
Added four methods to the 'PCM' class to allow users to get detailed information about the device:

  - `getformats()` returns a dictionary of name / value pairs, one for each of the card's
    supported formats - e.g. `{"U8": 1, "S16_LE": 2}`,
  - `getchannels()` returns a list of the supported channel numbers, e.g. `[1, 2]`,
  - `getrates()` returns supported sample rates for the device, e.g. `[48000]`,
  - `getratebounds()` returns the device's official minimum and maximum supported 
    sample rates as a tuple, e.g. `(4000, 48000)`.

The reason was simply to provide easy access to some of the very detailed information about a device, that's hidden in the `snd_pcm_hw_params_t` object.  I was writing a python application using this library that needed to introspect the device, and I'm sure other people will find this useful too.

### CODE REVIEW

File: alsaaudio.c [hunk 1]
```diff
@@ -572,6 +630,169 @@ alsapcm_dumpinfo(alsapcm_t *self, PyObject *args)
     return Py_None;
 }
 
+// auxiliary function
+
+
+static PyObject *
+alsapcm_getformats(alsapcm_t *self, PyObject *args) {
+	snd_pcm_t *pcm = self->handle;
+		if (!pcm) {
```
Comment (larsimmisch) [inline alsaaudio.c]: Inconsistent whitespace?

Comment (jdstmporter) [reply alsaaudio.c]: Yes - my Eclipse editor was having a very strange day, an the indentations were all over the place.  Sorry.  Can fix if required.

I'll have a think about tests - probably can't just be pure Python.

File: alsaaudio.c [hunk 2]
```diff
@@ -572,6 +630,169 @@ alsapcm_dumpinfo(alsapcm_t *self, PyObject *args)
...
+		snd_pcm_hw_params_alloca(&params);
+		int err = snd_pcm_hw_params_any(pcm, params);
+		if (err < 0) {
+			PyErr_SetString(ALSAAudioError, "Cannot get hardware parameters");
+			return NULL;
+		}
+
+	PyObject *fmts = PyDict_New();
+	for (int i = 0; i < ARRAY_SIZE(ALSAFormats); ++i) {
+		snd_pcm_format_t format = ALSAFormats[i];
```
Comment (larsimmisch) [inline alsaaudio.c:653]: `i` should be `size_t` (fixed locally)

File: alsaaudio.c [hunk 3]
```diff
@@ -572,6 +630,169 @@ alsapcm_dumpinfo(alsapcm_t *self, PyObject *args)
...
+	}
+	else if(!snd_pcm_hw_params_test_rate(pcm, params, min + 1, 0)) {
+		PyObject *minp=PyLong_FromLong(min);
+		PyObject *maxp=PyLong_FromLong(max);
+		return PyTuple_Pack(2,minp,maxp);
+	}
+	else {
+		PyObject *rates=PyList_New(0);
+		for(int i=0;i<ARRAY_SIZE(ALSARates);i++) {
+			unsigned rate=ALSARates[i];
```
Comment (larsimmisch) [inline alsaaudio.c:736]: `size_t` again.

**Pull request merged**